tests/CMakeLists.txt

```diff
@@ -1,21 +1,156 @@
+include(CMakePrintHelpers)
 
-include_directories(${CMAKE_BINARY_DIR} ${CMAKE_SOURCE_DIR}/include)
+option(PRESSIO_ENABLE_CXX14 "Enable C++14" OFF)
```

Comment:
@fnrizzi Can we actually drop support for C++14? I can see that Eigen defaults to C++14, but Trilinos, Kokkos (and possibly more TPLs) require C++17. 
We could add that to backlog for later.

Reply:
cannot drop 14 yet, let's add it to backlog!